### PR TITLE
Issue 3/app.js

### PR DESCRIPTION
**Pull request by AlexandrVinogradov**
**Path:** [`src/App.js`](https://github.com/AlexandrVinogradov/vending-machine-react/blob/master/src/App.js)

Link #3

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -33,75 +26,17 @@ class App extends React.Component {
...
-        isError: true,
-        errorMessageUnknowBanknote: 'Unknow Banknote',
-      })
-    }
-  }
-  balanceInputValue = e => {
+  setValuesOfInsertBanknoteForm = balance => {
     this.setState({
-      balanceInputValue: parseInt(e.currentTarget.value),
+      balance: balance,
```
Comment (sevenns) [inline src/App.js]: You can use another way, with es6:
```
{
  prop: "value",
  balance,
}
```

File: src/App.js [hunk 2]
```diff
@@ -33,75 +26,17 @@ class App extends React.Component {
...
-        isError: true,
-        errorMessageIncorrect: 'Enter correct number',
-      })
-    }
-  }
-  productInputValue = e => {
+  setValuesOfChooseProductForm = (selectedProduct, coins, change) => {
     this.setState({
-      productInputValue: parseInt(e.currentTarget.value),
+      selectedProduct: selectedProduct,
```
Comment (sevenns) [inline src/App.js]: Again here

File: src/componets/ChooseProductForm/ChooseProductForm.jsx [hunk 3]
```diff
@@ -1,6 +1,70 @@
...
+      this.props.products.some(p => p.id === this.state.productInputValue) &&
+      this.props.balance >= this.props.products.find(p => p.id === this.state.productInputValue).price
+    ) {
+      this.setState(
+        {
+          selectedProduct: parseInt(this.state.productInputValue),
+          errorMessageIncorrect: '',
+          coins: localCoins,
+          change:
+            parseInt(this.props.balance) - this.props.products.find(p => p.id === this.state.productInputValue).price,
```
Comment (sevenns) [inline src/componets/ChooseProductForm/ChooseProductForm.jsx]: What if the product is not found?

File: src/App.js (hunk 1 above)
Comment (AlexandrVinogradov) [reply src/App.js]: done
Link: #6
https://github.com/AlexandrVinogradov/vending-machine-react/blob/f2919431799bf73ed82a1ddbed6d3fc9ee8c981a/src/App.js#L31

File: src/App.js (hunk 2 above)
Comment (AlexandrVinogradov) [reply src/App.js]: done
Link: #6

https://github.com/AlexandrVinogradov/vending-machine-react/blob/f2919431799bf73ed82a1ddbed6d3fc9ee8c981a/src/App.js#L34

**Pull request merged**